### ISSUE TITLE
Limit MAX_DATA to the integer range (64 or 62 depending..)

### ISSUE THREAD

**Issue by mcmanus** [quicwg/base-drafts#873]
dimitry and @martinthomson  had a bit of a discussion on this in August: https://www.ietf.org/mail-archive/web/quic/current/msg02117.html

tl;dr - max-data is effectively a 74 bit integer.. plausibly much bigger than required, extremely hard to do integration testing with so presumably buggy as its > common native types.. at terrabit rates it would take years to transfer 64/62 bits (the obvious alternative) worth of flow controlled bytes.

I think this is an architectural issue because the bug risk for interop failure in long deployed systems is significant.

The existing arrangement falls short of accommodating the full sized packets across the whole packet number space anyhow. (to say nothing of dreaming of bigger MTUs) If we're going to keep it supersized then +16 bits makes a lot more sense to me than +10.
